Docs/reference/content/reference/driver/crud/client_side_encryption.md

```diff
@@ -181,4 +181,153 @@ namespace MongoDB.Driver.Examples
+using System.Threading;
+using MongoDB.Bson;
+using MongoDB.Driver.Encryption;
+using MongoDB.Libmongocrypt;
+
+namespace MongoDB.Driver.Examples
+{
+    public class ExplicitEncryptionAndAutoDecryptionExamples
+    {
+        private const string LocalMasterKey = "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk";
+
+        public static void Main(string[] args)
+        {
+            var localMasterKey = Convert.FromBase64String(LocalMasterKey);
+            var kmsProviders = new Dictionary<string, IReadOnlyDictionary<string, object>>();
+            var localKey = new Dictionary<string, object>
+            {
+                { "key", localMasterKey }
+            };
+            kmsProviders.Add("local", localKey);
+
+            var keyVaultNamespace = CollectionNamespace.FromFullName("admin.datakeys");
+            var collectionNamespace = CollectionNamespace.FromFullName("test.coll");
+            var autoEncryptionOptions = new AutoEncryptionOptions(
+                keyVaultNamespace,
+                kmsProviders,
+                bypassAutoEncryption: true);
+            var clientSettings = new MongoClientSettings
+            {
+                AutoEncryptionOptions = autoEncryptionOptions,
+            };
+            var mongoClient = new MongoClient(clientSettings);
+            var database = mongoClient.GetDatabase(collectionNamespace.DatabaseNamespace.DatabaseName);
+            database.DropCollection(collectionNamespace.CollectionName);
+            var collection = database.GetCollection<BsonDocument>(collectionNamespace.CollectionName);
+
+            var keyVaultClient = new MongoClient();
+            var keyVaultDatabase = keyVaultClient.GetDatabase(keyVaultNamespace.DatabaseNamespace.DatabaseName);
+            keyVaultDatabase.DropCollection(keyVaultNamespace.CollectionName);
+
+            // Create the ClientEncryption instance
+            var clientEncryptionSettings = new ClientEncryptionOptions(
+                keyVaultClient,
+                keyVaultNamespace,
+                kmsProviders);
+            var clientEncryption = new ClientEncryption(clientEncryptionSettings);
+
+            var dataKeyId = clientEncryption.CreateDataKey(
+                "local",
+                new DataKeyOptions(),
+                CancellationToken.None);
+
+            var originalString = "123456789";
+            Console.WriteLine($"Original string {originalString}.");
+
+            // Explicitly encrypt a field
+            var encryptOptions = new EncryptOptions(
+                EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic.ToString(),
+                keyId: dataKeyId);
+            var encryptedFieldValue = clientEncryption.Encrypt(
+                originalString,
+                encryptOptions,
+                CancellationToken.None);
+            Console.WriteLine($"Encrypted value {encryptedFieldValue}.");
```

Comment:
encryptedFieldValue is a BsonBinaryData value.

Does the implied call to ToString produce acceptable results?

Reply:
The target of this call is to clearly prove that the encrypted result is not the same as the original string. So, taking this into account it looks acceptable. But since we called Console.WriteLine only in the last test lines in the previous version (without intermediate results), I'm ok to remove this line at all

Reply:
I'm all for the presence of the Console.WriteLine (to clearly prove that the encrypted result is not the same).

My question was just about the **formatting** of the output.

Don't remove the line.

Reply:
> the formatting of the output.

which format do you expect? I would say that just call ToString(`implicitly`) is enough for the target of this example

Reply:
I was just asking if you had looked at the format and where satisfied that it was useful output.

We can leave it as is.